From 072d27fa368246a9c2bfbd8ba9f9c662ecda0b88 Mon Sep 17 00:00:00 2001
Message-Id: <072d27fa368246a9c2bfbd8ba9f9c662ecda0b88.1555511658.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Wed, 17 Apr 2019 16:29:57 +0200
Subject: [PATCH] trivial: Fix error checking for sd_pid_get_session and
 sd_session_get_seat

These functions can return a positive integer on success. Fix the
conditions to only regard negative return values as errors.

Fixes commit 6ffe2f6aae4bdc8331cab1fd9379ed8aef759749.
Fixes https://bugs.archlinux.org/task/37014
---
 src/cd-main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/cd-main.c b/src/cd-main.c
index 514ae5c..ffcecb8 100644
--- a/src/cd-main.c
+++ b/src/cd-main.c
@@ -491,15 +491,15 @@ cd_main_get_seat_for_process (guint pid)
 
 	/* get session the process belongs to */
 	rc = sd_pid_get_session (pid, &sd_session);
-	if (rc != 0) {
+	if (rc < 0) {
 		g_warning ("failed to get session [pid %u]: %s",
 			   pid, strerror (-rc));
 		goto out;
 	}
 
 	/* get the seat the session is on */
 	rc = sd_session_get_seat (sd_session, &sd_seat);
-	if (rc != 0) {
+	if (rc < 0) {
 		g_warning ("failed to get seat for session %s [pid %u]: %s",
 			   sd_session, pid, strerror (-rc));
 		goto out;
-- 
2.21.0

